### PR TITLE
Add alternative server with the same interface as HEASARC, but different tables (more recent data of INTEGRAL, some CTA tables)

### PR DESCRIPTION
**Pull request by volodymyrss**
[ISDC](https://www.isdc.unige.ch/) is providing INTEGRAL data for the community, and HEASARC imports tables from it, with a delay. ISDC uses [copy of the HEASARC service](https://www.isdc.unige.ch/browse/w3table.pl) - older version, and hence can be added as an alternative source, with some extra handling of exception in decoding, relevant in some cases.

Relation between ISDC and HEASARC is explained here: https://heasarc.gsfc.nasa.gov/docs/integral/inthp_archive.html

### CODE REVIEW

**Comment by volodymyrss** [comment]
In addition to ISDC interface, we suggest a modification to validate requests, forbidding unknown parameters, which are otherwise ignored.  Without this, a simple typo could result in unexpected results, without a warning.

**Comment by volodymyrss** [comment]
It is possible that instead it would be better to make another module, say, "isdc",  and re-use most of the heasarc module in it. Though most of the changes to heasarc module here (such as allowed parameter checking, time range) could be  of general use.

**Comment by volodymyrss** [comment]
Thanks @bsipocz for assigning the label!
I wonder if I am following the correct procedure with this PR, by creating it without prior issue or other discussions. I would not want to break conventions and appear inconsiderate.

File: astroquery/heasarc/core.py
Comment (keflavich) [inline astroquery/heasarc/core.py:210]: `content.decode` should (almost) never be needed; you may want to use `response.text` instead of `response.content` when receiving the web content.

File: astroquery/heasarc/core.py
Comment (keflavich) [inline astroquery/heasarc/core.py:249]: following the note above - maybe this should be `response.text`?

File: docs/heasarc/heasarc.rst
Comment (keflavich) [inline docs/heasarc/heasarc.rst:193]:
```suggestion
---------------------------------
```

File: docs/heasarc/heasarc.rst
Comment (keflavich) [inline docs/heasarc/heasarc.rst:197]: suggestion: add an example grabbing one of those new tables to show users why they might want to use this mirror

File: astroquery/heasarc/core.py
Comment (keflavich) [inline astroquery/heasarc/core.py:51]: is this definitely right?  `Entry` is used below, but you changed to `entry` here.  It might be better to use the same capitalization everywhere for consistency; maybe it's better to change all to `Entry`, since that appears to be what the service expects?

**Review by keflavich — CHANGES_REQUESTED** [review]
@volodymyrss Thanks for the PR, and sorry it took us a long time to see it.  I have a few minor suggestions, but this looks like a straightforward and helpful addition.

**Comment by keflavich** [comment]
@volodymyrss you followed the right procedure - we welcome PRs like this one!  It does help to remind us if we haven't gotten to it; we don't mind an occasional "bump".

File: astroquery/heasarc/core.py
Comment (volodymyrss) [reply astroquery/heasarc/core.py:51]: The `_args_to_payload` function actually documents `entry`:

https://github.com/astropy/astroquery/blob/554d35fee2f1339b90a75862944c968c8eda61e4/astroquery/heasarc/core.py#L241-L245

and also actually uses it:

https://github.com/astropy/astroquery/blob/554d35fee2f1339b90a75862944c968c8eda61e4/astroquery/heasarc/core.py#L283-L286

Passing `Entry` was silently ignored since there was no check for unused parameters. I guess it was just a typo, and the developers anticipated to use `entry`.
It had no impact though, since in this particular call it is set to `none` which is the default when it is not set at all.

Note in the quote above that `action` function argument is similarly transformed to `Action` for the service.
This way the function arguments are always lowercase, and are transformed to different cases as expected by the service, which depends on the argument.

I could certainly change it so that Entry and Action are always capitalized.
Then I would probably also change other parameters, and descriptions.
Things like `radius`:

https://github.com/astropy/astroquery/blob/554d35fee2f1339b90a75862944c968c8eda61e4/astroquery/heasarc/core.py#L325-L327

But what worries me is that this will change user interface:

https://github.com/astropy/astroquery/blob/554d35fee2f1339b90a75862944c968c8eda61e4/astroquery/heasarc/tests/test_heasarc_remote.py#L68

Which is something I was specifically trying to avoid. Would not want to break somebody's (and our own) scripts.

So let me know what you prefer! 
Should I change all arguments, including those provided to the user, to match those expected by the service?
Only change some?

Comment (keflavich) [reply astroquery/heasarc/core.py:51]: You raise great points here.

OK, let's stick with the lowercase for user-facing parameters, and therefore a mix of cases throughout the code (unfortunately).  It may be useful to include a comment briefly summarizing why we're using mixed case - i.e., just summerize what you said above, but only in 1 line.

File: docs/heasarc/heasarc.rst
Comment (volodymyrss) [reply docs/heasarc/heasarc.rst]: good idea, added. And I also added example for requesting time range in general case (as well as in our tables - it is critical there).

File: astroquery/heasarc/core.py
Comment (volodymyrss) [reply astroquery/heasarc/core.py]: ok, makes sense, I tried to avoid decode here and in other places where response.text can be used instead.

File: astroquery/heasarc/core.py
Comment (volodymyrss) [reply astroquery/heasarc/core.py]: ok, done.

**Comment by volodymyrss** [comment]
Thanks! Please let me know there is anything else I could improve. 

I looked into the non-remote, I could add some, but I am not entirely sure they are useful, since they will not break if the service changes. But the change at the service will render module not useful, and it should be noticed.
Maybe especially in this case, service output post-processing is virtually non-existent, and what is valuable is the interface.

**Comment by volodymyrss** [comment]
Hello @keflavich , as suggested I bump!

File: astroquery/heasarc/core.py
Comment (keflavich) [inline astroquery/heasarc/core.py:5]: we shouldn't use `_`-preceded tools - they are private and therefore are not guaranteed to be available

File: astroquery/heasarc/core.py
Comment (keflavich) [inline astroquery/heasarc/core.py:186]: is there another way to achieve this without using a private function?

**Review by keflavich — CHANGES_REQUESTED** [review]
Just one change left before I approve - if possible, we should avoid using private functions from astropy.

File: astroquery/heasarc/core.py
Comment (volodymyrss) [reply astroquery/heasarc/core.py]: Looks like there is a simpler way indeed, just:

```python
 c.format = str(c.format).replace("I", "A")
```

It seems to work the same, I added a small extra test to make sure it is good.

Not sure why I did not use this approach to begin with, it must have not worked, it was some time ago.

**Review by keflavich — APPROVED** [review]
@bsipocz @ceb8 either one of you want to give a final once-over and merge?

**Comment by ceb8** [comment]
Thanks @volodymyrss !

**Comment by volodymyrss** [comment]
Thanks to you all!

**Pull request merged**
